undertow/src/main/java/org/wildfly/extension/undertow/security/jaspi/JASPIAuthenticationMechanism.java

```diff
@@ -135,7 +135,7 @@ public AuthenticationMechanismOutcome authenticate(final HttpServerExchange exch
                 cache = Boolean.valueOf((String)registerObj);
             }
             sc.authenticationComplete(authenticatedAccount, authType, cache);
-        } else if (isValid && authenticatedAccount == null && !isMandatory(requestContext)) {
+        } else if (!isMandatory(requestContext)) {
             outcome = AuthenticationMechanismOutcome.NOT_ATTEMPTED;
```

Comment:
So we could either disable it by default for JAAS, or just disable it altogether. I'm leaning towards just disabling it by default. I am starting to think that the benefit it brings could be outweighed by the potential support cases it could generate when auth does not behave as people expect.